modules/core/core-api/src/main/java/com/enonic/xp/vfs/VirtualFilePathImpl.java

```diff
@@ -110,7 +111,7 @@ String join()
     @Override
     public String getName()
     {
-        return this.elements.getLast();
+        return this.elements.get(this.elements.size() - 1);
```

Comment:
Need to restore getLast to restore performance

Reply:
well, it is almost same.  because linked list is smart enough to iterate from tail in this case.